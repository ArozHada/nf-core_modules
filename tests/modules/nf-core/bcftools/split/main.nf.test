nextflow_process {

    name "Test Process BCFTOOLS_SPLIT"
    script "modules/nf-core/bcftools/split/main.nf"
    process "BCFTOOLS_SPLIT"
    config "tests/modules/nf-core/bcftools/split/nextflow.config"
    tag "bcftools_split"

    test("Human Illumina VCF") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['homo_sapiens']['illumina']['test_genome_vcf_gz'], checkIfExists: true),
                            file(params.test_data['homo_sapiens']['illumina']['test_genome_vcf_gz_tbi'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path("$outputDir/bcftools/test.chr22.vcf.gz").linesGzip[95-226],
                                  path("$outputDir/bcftools/versions.yml")).match() }
            )
        }
    }
}
