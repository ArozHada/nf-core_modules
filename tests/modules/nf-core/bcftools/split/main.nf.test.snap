{
    "Human Illumina VCF": {
        "content": [
            "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\ttestN",
            "versions.yml:md5,25d39fb2e3741d7800cd88a4c144840f"
        ],
        "timestamp": "2023-04-11T00:54:57+0000"
    }
}